### DISCUSSION TITLE
Some additional clarification with DuckDNS in mind

### DISCUSSION THREAD

**Discussion by Kureigu07** [llamafilm/tesla-http-proxy-addon#74] [General]
Please have the official [DOCS](https://github.com/llamafilm/tesla-http-proxy-addon/blob/main/tesla_http_proxy/DOCS.md) open to refer to.

Assuming you already have NGINX Home Assistant SSL proxy and DuckDNS set up with Let's Encrypt enabled in the config and can access your Home Assistant instance via it;

Log in to duckdns.org and create a unique subdomain for the Tesla HTTP Proxy i.e. mytesla.duckdns.org
Check your new domain with https://dnschecker.org/, and do not continue until this passes most, ideally all. (This can take a while)
Add your tesla domain to the domains section of the DuckDNS add-on, it will restart upon saving. Check the log to ensure the SSL certificates are valid:

> + Responding to challenge for mytesla.duckdns.org authorization...
>  + Challenge is valid!

If the log shows domain conflict or doesn't pass the challenge, try restarting the DuckDNS add-on.

Next configure "NGINX Home Assistant SSL proxy" as shown in the DOCS, likely just requiring the addition of the customize section.
Save and restart the add-on, the log should show no errors.

Then configure the Tesla HTTP Proxy add-on as described in the DOCS. Noting that the FQDN should be your new tesla specific address.
If you have already run this add-on before this point, you will need to delete [these files](https://github.com/llamafilm/tesla-http-proxy-addon/wiki/Read-this-before-asking-for-help#troubleshooting-steps) the easiest way is likely with the terminal add-on with the following commands

```
rm /share/nginx_proxy/nginx_tesla.conf
rm /share/tesla/*.*
```

Ensure "Regenerate Tesla authentication" is on and start the add-on.

View the logs, success will have the line "NOTICE: Request application access with Tesla, then fill in credentials and restart addon." in blue.

Continue as per the DOCS

**Comment by baylanger** [comment]
@Kureigu07 Can you edit your post, there's a space after the last `/` that shouldn't be there... and instead of using `*.*` , a single `*` does the same.

```
rm /share/tesla/ *.*
```

**Reply by baylanger** [reply]
FYI I quickly modded the WiKi page `Read this before asking for help` [Troubleshooting steps](https://github.com/llamafilm/tesla-http-proxy-addon/wiki/Read-this-before-asking-for-help#troubleshooting-steps) showing end-users how to delete the files.

**Comment by llamafilm** [comment]
@Kureigu07 I think this would make a nice wiki page, so it doesn't get buried in other discussions.  Would you like to make a page there? https://github.com/llamafilm/tesla-http-proxy-addon/wiki

**Reply by Kureigu07** [reply]
I've created a new [page ](https://github.com/llamafilm/tesla-http-proxy-addon/wiki/Setup-guide-using-existing-DuckDNS-&-NGINX-Home-Assistant-SSL-proxy), hopefully it is up to standard
